lib/metadataTypes/MobileMessage.js

```diff
@@ -0,0 +1,550 @@
+                } else if (Array.isArray(temp.entry)) {
+                    iteratorField = 'entry';
+                } else {
+                    throw new TypeError('Could not find an array to iterate over');
+                }
+            }
+            if (collector && Array.isArray(temp[iteratorField])) {
+                collector[iteratorField].push(...temp[iteratorField]);
+            } else if (collector == null) {
+                collector = temp;
+            }
+            if (
+                Array.isArray(collector[iteratorField]) &&
+                collector[iteratorField].length >= temp[countName] &&
+                (!isTransactionalMessageApi ||
+                    (isTransactionalMessageApi && temp[countName] != temp[pageSizeName]))
+            ) {
+                // ! the transactional message API returns a value for "count" that represents the currently returned number of records, instead of the total amount. checking for count != pageSize is a workaround for this
+                // * opened Support Case #43988240 for this issue
+                shouldPaginate = false;
+            } else {
+                page++;
+                shouldPaginate = true;
+                if (this.options?.eventHandlers?.onLoop) {
+                    this.options.eventHandlers.onLoop(null, collector?.[iteratorField]);
+                }
+            }
+        } while (shouldPaginate);
+        return collector;
+    }
+
+    /**
+     * Updates a single item
+     *
+     * @param {TYPE.MetadataTypeItem} metadata a single item
+     * @returns {Promise} Promise
+     */
+    static update(metadata) {
+        return super.updateREST(
+            metadata,
+            '/legacy/v1/beta/mobile/message/' + metadata[this.definition.idField],
+            'post' // upsert API, post for insert and update!
+        );
+    }
+
+    /**
+     * Creates a single item
+     *
+     * @param {TYPE.MetadataTypeItem} metadata a single item
+     * @returns {Promise} Promise
+     */
+    static create(metadata) {
+        return super.createREST(metadata, '/legacy/v1/beta/mobile/message/');
+    }
+    /**
+     * helper for {@link preDeployTasks} that loads extracted code content back into JSON
+     *
+     * @param {TYPE.MetadataTypeItem} metadata a single definition
+     * @param {string} deployDir directory of deploy files
+     * @param {string} [templateName] name of the template used to built defintion (prior applying templating)
+     * @returns {Promise.<string>} code
+     */
+    static async _mergeCode(metadata, deployDir, templateName) {
+        const fileExtension = 'amp';
+        templateName = templateName || metadata[this.definition.keyField];
+        let code;
+        const codePath = File.normalizePath([
+            deployDir,
+            this.definition.type,
+            templateName + '.' + this.definition.type + '-meta',
+        ]);
+
+        if (await File.pathExists(codePath + '.' + fileExtension)) {
```

Comment:
why declare "code"? you can just return from code = await below directly

Reply:
good point